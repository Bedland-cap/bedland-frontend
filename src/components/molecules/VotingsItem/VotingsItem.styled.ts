import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const VotingsItemContainer = styled.div<{ palette: PaletteType }>`
  align-items: center;
  border-bottom: 1px solid ${({ palette }) => palette.sidebarTabs};
  display: flex;
  margin-right: 1.1rem;
  padding: 2rem 1.939rem 2rem 2.313rem;
  :last-child {
    margin-bottom: 0.938rem;
  }
`;

export const IconWrapper = styled.div`
  transform: rotate(90deg);
`;

export default VotingsItemContainer;
