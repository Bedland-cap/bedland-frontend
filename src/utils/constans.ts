export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
