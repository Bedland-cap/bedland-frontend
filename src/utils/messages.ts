const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
