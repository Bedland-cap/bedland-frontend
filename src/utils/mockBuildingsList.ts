import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
