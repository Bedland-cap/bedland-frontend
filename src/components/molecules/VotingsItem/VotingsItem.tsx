import Icon from 'components/atoms/Icon/Icon';
import IconButton from 'components/atoms/IconButton/IconButton';
import Typography from 'components/atoms/Typography/Typography';
import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './VotingsItem.styled';
import { dateDiffinDays, addLeadingZero, switchVariant } from './VotingsItem.utils';

export type VotingsItemProps = {
  deadlineDate: Date;
};

const VotingsItem = ({ deadlineDate }: VotingsItemProps) => {
  const { palette } = useContext(ThemeContext);

  const newDeadlineDate = new Date(deadlineDate);
  const curDate = new Date();

  const dataForm = `${addLeadingZero(
    newDeadlineDate.getMonth() + 1,
  )}/${newDeadlineDate.getFullYear()}`;

  return (
    <Styled.VotingsItemContainer palette={palette}>
      <Icon name='document' color='inputGrey' size={20} isActive={false} />
      <Typography
        variant='header4'
        style={{
          marginLeft: '1.188rem',
          flex: 1,
          fontWeight: dateDiffinDays(curDate, newDeadlineDate) < 0 ? '400' : '600',
        }}
        color='inputGrey'
      >
        Resolution {dataForm}
      </Typography>
      <Typography
        variant='header4'
        color='inputGrey'
        style={{ fontWeight: dateDiffinDays(curDate, newDeadlineDate) < 0 ? '400' : '600' }}
      >
        {switchVariant(curDate, newDeadlineDate)}
      </Typography>
      <Styled.IconWrapper>
        <IconButton icon='arrowRight' color='inputGrey' size={20} onClick={() => null} />
      </Styled.IconWrapper>
    </Styled.VotingsItemContainer>
  );
};

export default VotingsItem;
