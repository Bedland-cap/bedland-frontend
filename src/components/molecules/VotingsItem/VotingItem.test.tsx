import { renderWithProviders } from 'utils/test.utils';
import Badge from 'components/atoms/Badge/Badge';
import VotingsItem, { VotingsItemProps } from './VotingsItem';
import { addLeadingZero, dateDiffinDays, switchVariant } from './VotingsItem.utils';

describe('VotingsItem', () => {
  const item = ({ deadlineDate }: VotingsItemProps) =>
    renderWithProviders(<VotingsItem deadlineDate={deadlineDate} />, {});

  it('should renders correctly VotingsItem', () => {
    expect(
      item({
        deadlineDate: new Date(),
      }),
    ).toMatchSnapshot();
  });
  it('dateDiffinDays should calculate correctly positive difference between two dates', () => {
    const result = dateDiffinDays(new Date(2023, 3, 19), new Date(2023, 3, 21));
    expect(result).toEqual(2);
  });
  it('dateDiffinDays should calculate correctly negative difference between two dates', () => {
    const result = dateDiffinDays(new Date(2023, 3, 21), new Date(2023, 3, 19));
    expect(result).toEqual(-2);
  });

  it('addLeadingZero should add 0 at the beginning when result is less than 10 ', () => {
    const result = addLeadingZero(new Date(2023, 3, 5).getMonth() + 1);
    expect(result).toEqual('04');
  });
  it('addLeadingZero shouldnt add 0 at the beginning when result is bigger than 10 ', () => {
    const result = addLeadingZero(new Date(2023, 10, 5).getMonth() + 1);
    expect(result).toEqual('11');
  });
  it('switch variant switches correctly variants', () => {
    const result = switchVariant(new Date(2023, 10, 5), new Date(2023, 10, 6));
    expect(result).toEqual(
      <Badge variant='danger'>
        {dateDiffinDays(new Date(2023, 10, 5), new Date(2023, 10, 6))} left
      </Badge>,
    );
  });
});
