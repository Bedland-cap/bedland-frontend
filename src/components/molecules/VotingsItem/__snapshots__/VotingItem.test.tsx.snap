// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VotingsItem should renders correctly VotingsItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FF4438;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.688rem;
  width: 5.875rem;
}

.c3 {
  background-color: 1px solid #FF4438;
  border: 1px solid transparent;
  color: #fff;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F56513;
  border: 1px solid #F56513;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  width: 100%;
}

.c7:hover {
  background-color: #FF7C33;
  border: 1px solid #FF7C33;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0:last-child {
  margin-bottom: 0.938rem;
}

.c1 {
  border-bottom: none;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: 1.1rem;
  padding: 2rem 1.939rem 2rem 2.313rem;
}

.c1:last-child {
  margin-bottom: 0rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.875rem;
  height: 0px;
  opacity: 0;
  overflow: hidden;
  -webkit-transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  visibility: hidden;
  width: 35.25rem;
}

.c4 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c0 c1"
          data-testid="votings-item"
        >
          <svg
            fill="none"
            height="20"
            viewBox=" 0 0 20 19"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-document"
            />
            <path
              d="M17 17.5H3C1.4 17.5 1 16.167 1 15.5V3.5C1 2.96957 1.21071 2.46086 1.58579 2.08579C1.96086 1.71071 2.46957 1.5 3 1.5H13C13.5304 1.5 14.0391 1.71071 14.4142 2.08579C14.7893 2.46086 15 2.96957 15 3.5V9.5M17 17.5C16.333 17.5 15 17.1 15 15.5V9.5M17 17.5C18.6 17.5 19 16.167 19 15.5V9.5H15"
              stroke="#4B4B4B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
          <h4
            style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); margin-left: 1.188rem; flex: 1;"
          >
            Resolution 
            03/2023
          </h4>
          <h5
            style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            <div
              class="c2 c3"
            >
              0
               left
            </div>
          </h5>
          <div
            class="c4"
            data-testid="votings-item-icon"
          >
            <button
              class="c5"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-arrowRight"
                />
                <path
                  d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                  stroke="#4B4B4B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.39167"
                />
              </svg>
              <h4
                style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
              />
            </button>
          </div>
        </div>
        <div
          class="c6"
          data-testid="unfolded-container"
        >
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); padding: 10px 6px;"
          >
            Resolution on the transfer of the water supply network located in the housing estate to the municipal water supply and sewage company
          </p>
          <div
            class=""
            style="display: flex; justify-content: flex-end; width: 100%;"
          >
            <div
              class=""
            >
              <button
                class="c7"
                type="button"
              >
                <h4
                  style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(252, 254, 255);"
                >
                  <h5
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0;"
                  >
                    Vote now
                  </h5>
                </h4>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FF4438;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.688rem;
  width: 5.875rem;
}

.c3 {
  background-color: 1px solid #FF4438;
  border: 1px solid transparent;
  color: #fff;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F56513;
  border: 1px solid #F56513;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  width: 100%;
}

.c7:hover {
  background-color: #FF7C33;
  border: 1px solid #FF7C33;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0:last-child {
  margin-bottom: 0.938rem;
}

.c1 {
  border-bottom: none;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: 1.1rem;
  padding: 2rem 1.939rem 2rem 2.313rem;
}

.c1:last-child {
  margin-bottom: 0rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.875rem;
  height: 0px;
  opacity: 0;
  overflow: hidden;
  -webkit-transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  visibility: hidden;
  width: 35.25rem;
}

.c4 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c0 c1"
        data-testid="votings-item"
      >
        <svg
          fill="none"
          height="20"
          viewBox=" 0 0 20 19"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-document"
          />
          <path
            d="M17 17.5H3C1.4 17.5 1 16.167 1 15.5V3.5C1 2.96957 1.21071 2.46086 1.58579 2.08579C1.96086 1.71071 2.46957 1.5 3 1.5H13C13.5304 1.5 14.0391 1.71071 14.4142 2.08579C14.7893 2.46086 15 2.96957 15 3.5V9.5M17 17.5C16.333 17.5 15 17.1 15 15.5V9.5M17 17.5C18.6 17.5 19 16.167 19 15.5V9.5H15"
            stroke="#4B4B4B"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
        <h4
          style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); margin-left: 1.188rem; flex: 1;"
        >
          Resolution 
          03/2023
        </h4>
        <h5
          style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
        >
          <div
            class="c2 c3"
          >
            0
             left
          </div>
        </h5>
        <div
          class="c4"
          data-testid="votings-item-icon"
        >
          <button
            class="c5"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-arrowRight"
              />
              <path
                d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                stroke="#4B4B4B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.39167"
              />
            </svg>
            <h4
              style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
            />
          </button>
        </div>
      </div>
      <div
        class="c6"
        data-testid="unfolded-container"
      >
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); padding: 10px 6px;"
        >
          Resolution on the transfer of the water supply network located in the housing estate to the municipal water supply and sewage company
        </p>
        <div
          class=""
          style="display: flex; justify-content: flex-end; width: 100%;"
        >
          <div
            class=""
          >
            <button
              class="c7"
              type="button"
            >
              <h4
                style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(252, 254, 255);"
              >
                <h5
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0;"
                >
                  Vote now
                </h5>
              </h4>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
